src/client/src/producer.rs

```diff
@@ -61,7 +61,36 @@ impl TopicProducer {
             addr = spu_client.config().addr(),
             "Connected to replica leader:"
         );
-        send_record_raw(spu_client, &replica, Some(key), value).await
+        let records = &[(Some(key), value)];
+        send_records_raw(spu_client, &replica, records).await?;
+        Ok(())
+    }
+
+    #[instrument(
+        skip(self, records),
+        fields(topic = %self.topic),
+    )]
+    pub async fn send_all<K, V, I>(&self, records: I) -> Result<(), FluvioError>
```

Comment:
since we are converting to `Vector` anyway, we should take `Vector` rather than `Iterator`.  This may solve API issue as well.

Reply:
I think that using `IntoIterator` should be just fine... Using `Vec` would not solve any of the open API rough-edges, it would actually make them worse. By using `IntoIterator` we can change the internal representation without breaking the API, but if we bake Vec into the type signature, any change would become a breaking change.

Reply:
Since we've recently decided to use `Bytes` as the internal representation of records, I think that using `Into<Vec<u8>>` as our parameters should actually be a good API here 👍